wave_vision/src/utils.cpp

```diff
@@ -27,15 +27,74 @@ void projection_matrix(const Mat3 &K, const Mat3 &R, const Vec3 &t, MatX &P) {
     P = K * extrinsics;
 }
 
-void convertKeypoint(const cv::KeyPoint &keypoint, Vec2 &vec_keypoint) {
-    vec_keypoint(0) = keypoint.pt.x;
-    vec_keypoint(1) = keypoint.pt.y;
+Vec2 convertKeypoint(const cv::KeyPoint &keypoint) {
+    Vec2 vec_keypoint(keypoint.pt.x, keypoint.pt.y);
+
+    return vec_keypoint;
+}
+
+Vec2 convertKeypoint(const cv::Point2f &keypoint) {
+    Vec2 vec_keypoint(keypoint.x, keypoint.y);
+
+    return vec_keypoint;
+}
+
+cv::Point2f convertKeypoint(const Vec2 &keypoint) {
+    cv::Point2f cv_keypoint((float) keypoint(0), (float) keypoint(1));
+
+    return cv_keypoint;
 }
 
-void convertKeypoints(const std::vector<cv::KeyPoint> &keypoints,
-                      std::vector<Vec2> &vec_keypoints) {
+std::vector<Vec2> convertKeypoints(const std::vector<cv::KeyPoint> &keypoints) {
+    std::vector<Vec2> vec_keypoints;
+
     for (const auto &k : keypoints) {
         vec_keypoints.emplace_back(k.pt.x, k.pt.y);
     }
+
+    return vec_keypoints;
+}
+
+std::vector<Vec2> convertKeypoints(const std::vector<cv::Point2f> &keypoints) {
+    std::vector<Vec2> vec_keypoints;
+
+    for (const auto &k : keypoints) {
+        vec_keypoints.emplace_back(k.x, k.y);
+    }
+
+    return vec_keypoints;
+}
+
+std::vector<cv::Point2f> convertKeypoints(const std::vector<Vec2> &keypoints) {
+    std::vector<cv::Point2f> cv_keypoints;
+
+    for (const auto &k : keypoints) {
+        cv_keypoints.emplace_back((float) k(0), (float) k(1));
+    }
+
+    return cv_keypoints;
+}
+
+std::vector<cv::Mat> readImageSequence(const std::string &path) {
```

Comment:
Mmm good point. I still think this could be a useful utility. Will revisit in the future :slightly_smiling_face: